client_core/src/com/mygdx/game/client_core/network/EndTurnService.java

```diff
@@ -0,0 +1,28 @@
+package com.mygdx.game.client_core.network;
+
+import com.github.czyzby.websocket.WebSocket;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+
+import javax.inject.Inject;
+
+@Log
+public class EndTurnService {
+
+  private final WebSocket socket;
+  private final PlayerInfo playerInfo;
+
+  @Inject
+  public EndTurnService(
+      @NonNull WebSocket socket,
+      @NonNull PlayerInfo playerInfo
+  ) {
+    this.socket = socket;
+    this.playerInfo = playerInfo;
+  }
+
+  public void endTurn() {
+    log.info("end turn request send");
+    socket.send("end_turn");
```

Comment:
czy nie lepszym rozwiazaniem byłoby żeby w tym miejscu gracz ustawiał sobie od razu że zakończyła się jego tura?

Reply:
@Szomeszp  tutaj pojawia się ten problem, że nie mamy zrobionego sposobu na to jak cofnąć zmiany po stronie klienta, jeśli serwer jednak zdecyduje, że "twoja tura jeszcze trwa" - ten problem jest trochę bardziej ogólny, bo tyczy się każdego takiego "rozkazu" wydanego przez klienta do serwera, więc najlepiej byłoby to przemyśleć i znaleźć ogólne rozwiązanie

w tym przypadku gracz akurat obiecuje, że nie będzie już dalej klikał, więc problem jest faktycznie prostszy, bo nie wprowadziłby zmian w stanie

Reply:
Skoro serwer decyduje o aktywacji gracza poprzez wysłanie wiadomości, to wydaje mi się sensowne, żeby też serwer o tym decydował, te dwie rzeczy są obsługiwane w jednym miejscu w kodzie, i to wydaje mi się okej